standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -686,6 +687,31 @@ public void updateTransactionStatistics(UpdateTransactionalStatsRequest req) thr
             .addValue("tableId", req.getTableId()), null);
   }
 
+  @Override
+  public boolean hasTransactionalResource(Set<String> dbTable) throws MetaException {
+    if (dbProduct.isDERBY()) {
+      return true;
+    }
```

Comment:
Does it mean that a txn has to be always opened if the backend db is derby? Even if there are no transactional tables in the statement.

Reply:
no one is using derby in production, but the main reason is that derby doesn't support multi-column IN construct. 
tested so far on Postgres, MySQL, Oracle

Reply:
maybe we should consider moving to H2 in tests

Reply:
changed the strategy, not relevant any more